src/Lib/package.cpp

```diff
@@ -109,7 +109,9 @@ void test_datadir(void) {
   return;
 }
 
-int main(void) {
+// main() must be defined with the args in this format, otherwise we may get an
+// "undefined reference to SDL_main"
+int main(int argc, char *argv[]) {
```

Comment:
This finally fixed the "undefined reference to SDL_main" and the build of the binary and tests complete. Both tests fail though. I haven't investigated yet.